### PR TITLE
OpenProject Importer

### PR DESCRIPTION
**Pull request by longinius**
This plugin imports your open and assigned work packages from [OpenProject](https://www.openproject.org). It can also update previous imported work packages based on time entries in a given time frame.

The plugin uses the [OpenProject API](https://www.openproject.org/docs/api/) to fetch the data. The OpenProject URL can be specified if a self-hosted version is used.

### CODE REVIEW

**Comment by tyme-app** [comment]
Hey @longinius, that's awesome. Thank you for this contribution. 🤗

Just a small nitpick: Maybe you could shorten the ID prefixes a bit, because the maximum ID length is 128 characters.

File: OpenProjectImporter/script.js [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+            this.createOrUpdateTask(workPackage);
+        }
+    }
+
+    /**
+     * Creates or updates an existing project in Tyme.
+     * @param {*} project Project from OpenProject.
+     */
+    createOrUpdateProject(project) {
+        const id = "openproject-project-" + project.id;
```
Comment (tyme-app) [inline OpenProjectImporter/script.js]: Just `openproject-` as a prefix would be sufficient for all entities.

Comment (longinius) [reply OpenProjectImporter/script.js]: Makes sense, I've adjusted the prefix for all entities.

**Comment by tyme-app** [comment]
Thank you @longinius. Merged. Awesome 🥳

**Pull request merged**